roachtest: refactor run/bench code

RaduBerinde opened the pull request:
This commit cleans up some of the clunky roachtest run code. We move
all run code and flags to a separate file. We switch run flags to use
global vars (like all other flags). We also extract out the run/bench
code from the command-wide `PersistentPreRunE`.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110744)
<!-- Reviewable:end -->

RaduBerinde committed:
roachtest: refactor run/bench code

This commit cleans up some of the clunky roachtest run code. We move
all run code and flags to a separate file. We switch run flags to use
global vars (like all other flags). We also extract out the run/bench
code from the command-wide `PersistentPreRunE`.

Epic: none
Release note: None

srosenberg reviewed (approved):
Much cleaner, thanks for the refactoring! Before merging, let's do a smoke test gce run with SELECT_PROBABILITY=0.3

RaduBerinde commented:
TFTR! Seems to be running fine https://teamcity.cockroachdb.com/viewLog.html?buildId=11864800&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=buildResultsDiv

RaduBerinde commented:
bors r+
